frontend/components/coupons/Coupon.tsx

```diff
@@ -0,0 +1,50 @@
+import { Stack } from '@mui/system';
+import { Typography } from '@mui/material';
+import { CouponBaseData } from '../../types/coupons';
+import { couponContainerSx, leftSectionSx, rightSectionSx } from '../../styles/components/coupons/couponStyles';
+import { QRCodeSVG } from 'qrcode.react';
+
+interface Props {
+  coupon: CouponBaseData;
+}
+
+const Coupon = ({ coupon }: Props) => {
+  const websiteUrl = window.location.host;
```

Comment:
I'm also unsure if this will cause any problems in NextJS deployments (where window can be undefined). Not very sure but it looks fine for now cuz coupons are generated using client side rendering logic, and window will always be available in that scenario.

Reply:
Oh I see, the alternative is to use a frontend env variable but I need to check this with Siva on how he plans to deploy